dbms/src/DataStreams/HashJoinProbeBlockInputStream.cpp

```diff
@@ -67,17 +68,109 @@ Block HashJoinProbeBlockInputStream::getHeader() const
 
 Block HashJoinProbeBlockInputStream::readImpl()
 {
-    if (probe_process_info.all_rows_joined_finish)
+    // if join finished, return {} directly.
+    if (squashing_transform.isJoinFinished())
```

Comment:
I'm a bit confused as to whether the over_limit_block is output when `squashing_transform.isJoinFinished()`.
Please add a unit test to ensure that.

Reply:
> I'm a bit confused as to whether the over_limit_block is output when `squashing_transform.isJoinFinished()`. Please add a unit test to ensure that.

join_finished will be true only input block is {} in appendBlock, at this time the over_limit_block have already push into result blocks and  will not be put into new value,  the final block contains all rows need return in this read().
I think the current tests can cover this case.

Reply:
> I'm a bit confused as to whether the over_limit_block is output when `squashing_transform.isJoinFinished()`. Please add a unit test to ensure that.

join_finished will be true only input block is {} in appendBlock, at this time the over_limit_block have already push into result blocks and will not be put into new value, then the final block contains all rows need return in this read().
I think the current tests can cover this case.